test/testsyncengine.cpp

```diff
@@ -586,6 +586,34 @@ private slots:
         QVERIFY(localFileExists("A/.hidden"));
         QVERIFY(fakeFolder.currentRemoteState().find("B/.hidden"));
     }
+
+    void testNoLocalEncoding()
+    {
+        auto utf8Locale = QTextCodec::codecForLocale();
+        if (utf8Locale->mibEnum() != 106) {
+            qWarning() << "Test only works for UTF8 locale";
+            return;
+        }
+
+        FakeFolder fakeFolder{ FileInfo::A12_B12_C12_S12() };
+        QVERIFY(fakeFolder.syncOnce());
+        QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
+
+        fakeFolder.remoteModifier().insert("A/tößt");
+        QVERIFY(fakeFolder.syncOnce());
+        QVERIFY(fakeFolder.currentLocalState().find("A/tößt"));
+
+        QTextCodec::setCodecForLocale(QTextCodec::codecForName("ASCII"));
```

Comment:
changing the locale after the sync means that the file was already synchronzed.
I would change the locale at the beginning of the test.


Also, while we are at it, it could be better to have a ISO-8859-15 locale,  and check that tößt is proberly synced,  but  tœβt or  🙈 is not

Reply:
That's intentional, I wanted to also verify that with an utf8 locale the file is synchronized as expected.

The intermediate locale sounds like a great idea.